conn_test.go

```diff
@@ -935,34 +944,44 @@ func TestNewSessionTimedOut(t *testing.T) {
 	require.Nil(t, session)
 
 	// should have one session to clean up
+	// TODO: sync with mux after abandoned session has been created
+	time.Sleep(100 * time.Millisecond)
 	require.Len(t, client.abandonedSessions, 1)
 	require.Len(t, client.sessionsByChannel, 1)
 
+	// we sleep here to wait for the begin performative to
+	// arrive, thus creating the now abandoned session.
+	// TODO: add test hooks to session mux to eliminate the sleep
```

Comment:
[Tracking issue](https://github.com/Azure/go-amqp/issues/299) for this and other TODOs added for augmented mux test hooks.